optimum/intel/ipex/inference.py

```diff
@@ -20,6 +22,51 @@
     import intel_extension_for_pytorch as ipex
 
 
+def prepare_jit_inputs(inputs, model_pipeline, dtype):
+    input_length = 1 if isinstance(inputs, str) else len(inputs)
+    if model_pipeline.tokenizer.pad_token is None:
+        model_pipeline.tokenizer.pad_token = model_pipeline.tokenizer.eos_token
+    dummy_input = model_pipeline.tokenizer.batch_encode_plus(inputs, return_tensors="pt", padding=True)
+    normalized_config = NormalizedConfigManager.get_normalized_config_class(model_pipeline.model.config.model_type)(
+        model_pipeline.model.config
+    )
+    num_block_layers = normalized_config.num_layers
+    num_attention_heads = normalized_config.num_attention_heads
+    num_embedding_size_per_head = normalized_config.hidden_size // normalized_config.num_attention_heads
+    if model_pipeline.model.config == "bloom":
+        past_key_values = tuple(
+            (
+                torch.zeros(int(num_attention_heads * input_length), num_embedding_size_per_head, 1).to(dtype),
+                torch.zeros(int(num_attention_heads * input_length), 1, num_embedding_size_per_head).to(dtype),
+            )
+            for _ in range(num_block_layers)
+        )
+    else:
+        past_key_values = tuple(
+            (
+                torch.zeros(input_length, num_attention_heads, 1, num_embedding_size_per_head).to(dtype),
+                torch.zeros(input_length, num_attention_heads, 1, num_embedding_size_per_head).to(dtype),
+            )
+            for _ in range(num_block_layers)
+        )
+    dummy_input["attention_mask"] = torch.cat(

```

Comment:
Why is this modification necessary ?

Reply:
We generated the past_key_values so the input sequence length increases, so the attention mask needs to increase its length to match the input length.